spring-cloud-config-client/src/test/java/org/springframework/cloud/config/client/ConfigClientPropertiesTests.java

```diff
@@ -59,7 +59,42 @@ public void explicitPassword() {
 		assertEquals("foo", locator.getUsername());
 		assertEquals("secret", locator.getPassword());
 	}
+	
+	@Test
+	public void testIfNoColonPresentInUriCreds() {
+		locator.setUri("http://foobar@localhost:9999");
+		locator.setPassword("secret");
+		assertEquals("http://localhost:9999", locator.getRawUri());
+		assertEquals("foobar", locator.getUsername());
+		assertEquals("secret", locator.getPassword());
+	}
 
+	@Test
+	public void testIfColonPresentAtTheEndInUriCreds() {
+		locator.setUri("http://foobar:@localhost:9999");
+		locator.setPassword("secret");
+		assertEquals("http://localhost:9999", locator.getRawUri());
+		assertEquals("foobar", locator.getUsername());
+		assertEquals("secret", locator.getPassword());
+	}
+	
+	@Test
+	public void testIfColonPresentAtTheStartInUriCreds() {
+		locator.setUri("http://:foobar@localhost:9999");
+		assertEquals("http://localhost:9999", locator.getRawUri());
+		assertEquals("user", locator.getUsername());
+		assertEquals("foobar", locator.getPassword());
+	}
+	
+	@Test
+	public void testIfColonPresentAtTheStartAndEndInUriCreds() {
+		locator.setUri("http://:foobar:@localhost:9999");
+		locator.setPassword("secret");
+		assertEquals("http://localhost:9999", locator.getRawUri());
+		assertEquals(":foobar", locator.getUsername());
```

Comment:
This test is also invalid the user name should be the empty string and the password should be `foobar:`